pkg/services/publicdashboards/database/database.go

```diff
@@ -200,10 +200,13 @@ func (d *PublicDashboardStoreImpl) GetPublicDashboard(ctx context.Context, orgId
 	pdRes := &PublicDashboard{OrgId: orgId, DashboardUid: dashboardUid}
 	err := d.sqlStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
 		// publicDashboard
-		_, err := sess.Get(pdRes)
+		exists, err := sess.Get(pdRes)
 		if err != nil {
 			return err
 		}
+		if !exists {
+			return ErrPublicDashboardNotFound
+		}
```

Comment:
I'm ok with this approach but we have other functions that are returning `nil` when there is no resource with the provided id. We should create a ticket to make all returns consistent in each layer (database and service) as part of the refactoring effort.

Reply:
I agree. I think in order to be as rest as we can, we should return an error when there's no resource. Maybe we should try to be consistent with this

Reply:
However, I don't like the way FE manages errors. You can see I had to hardcode the tooltip error to not be shown and I had to show it manually. with a nil response, it would be easier specially in this case where nil doesn't mean an error for the frontend, but there's no config already created.